crates/types/src/schema/service.rs

```diff
@@ -170,6 +197,83 @@ pub struct HandlerMetadata {
     #[serde(default, skip_serializing_if = "std::collections::HashMap::is_empty")]
     pub metadata: HashMap<String, String>,
 
+    /// # Idempotency retention
+    ///
+    /// The retention duration of idempotent requests for this service.
+    #[serde(
+        with = "serde_with::As::<Option<restate_serde_util::DurationString>>",
+        skip_serializing_if = "Option::is_none",
+        default
+    )]
+    #[cfg_attr(feature = "schemars", schemars(with = "Option<String>"))]
+    pub idempotency_retention: Option<Duration>,
```

Comment:
Is there a difference between `None` and `Some(Duration::Zero)`?

Reply:
I assume it is for deciding whether a value was explicitly set or not when merging the configurations from the manifest with some defaults.